src/index.rs

```diff
@@ -609,7 +609,7 @@ impl ResourceIndex {
                 "Any collision must involve at least 2 resources"
             );
             *collisions -= 1;
-            if *collisions <= 1 {
+            if *collisions == 1 {
                 self.collisions.remove(&old_id);
```

Comment:
Should we panic on negative values then?

Reply:
Yes. We should never reach this case

Reply:
It's of type `usize` so it would never be negative eitherway